deep4cast/forecasting.py

```diff
@@ -0,0 +1,224 @@
+# -*- coding: utf-8 -*-
+"""Time series regression module.
+
+This module provides access to forecasters that can be fit to univariate
+or multivariate time series.
+
+"""
+import numpy as np
+
+from abc import ABC, abstractmethod
+from hyperopt import hp, fmin, tpe
+from keras.optimizers import RMSprop, SGD
+from deep4cast.models import TemporalCNN, TruncatedRNN
+
+
+class Forecaster(ABC):
+    """Abstract data handler class."""
+
+    def __init__(self):
+        """Initialize properties."""
+        # Model-dependent internal variables
+        self._model = None
+        self._history = None
+
+        # Internal boolean checks
+        self.is_fitted = False
+
+    def fit(self, X, y, validation_split=0.0, verbose=0):
```

Comment:
This `Forecaster` class is designed for time series. So ideally, it should operate on time series data, without requiring users to do generate and normalize window data first.  Said differently, the class should own the window-data generation and normalization.

Reply:
I will include this as part of the module.